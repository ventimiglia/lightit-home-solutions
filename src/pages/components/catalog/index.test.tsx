import { render, fireEvent } from '@testing-library/react';
import Catalog from './';

const props = {
  title: 'Title',
  breadcrumb: 'Breadcrumb',
  onClick: () => {},
  src: 'image.jpg',
  name: 'Product Name',
}

describe('Catalog', () => {
  it('renders the catalog component', () => {
    const { container, getByText, getByAltText } = render(
      <Catalog>
        <Catalog.Title>{props.title}</Catalog.Title>
        <Catalog.Breadcrumb onClick={props.onClick}>{props.breadcrumb}</Catalog.Breadcrumb>
        <Catalog.Product src={props.src} name={props.name} />
      </Catalog>
    );

    const titleElement = getByText(props.title);
    expect(titleElement).toBeInTheDocument();

    const breadcrumbElement = getByText(props.breadcrumb);
    expect(breadcrumbElement).toBeInTheDocument();


    const productElement = getByText(props.name);
    expect(productElement).toBeInTheDocument();
    const imageElement = getByAltText(props.name);
    expect(imageElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("calls onClick event handlers when Breadcrumb are clicked", () => {
    const breadcrumbClickHandler = jest.fn();

    const {getByText} = render(
      <Catalog>
        <Catalog.Breadcrumb onClick={breadcrumbClickHandler}>{props.breadcrumb}</Catalog.Breadcrumb>
      </Catalog>
    );

    const breadcrumb = getByText(props.breadcrumb);
    fireEvent.click(breadcrumb);

    expect(breadcrumbClickHandler).toHaveBeenCalledTimes(1);
  });
});
