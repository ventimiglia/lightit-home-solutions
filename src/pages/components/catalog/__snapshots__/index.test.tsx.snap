// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog renders the catalog component 1`] = `
<div>
  <div
    class="w-full"
  >
    <h2
      class="text-left font-bold text-lg mb-4"
    >
      Title
    </h2>
    <p
      class="text-xs capitalize text-left w-full mb-2 cursor-pointer hover:underline"
    >
      Breadcrumb
    </p>
    <article>
      <img
        alt="Product Name"
        class="h-36 object-contain mb-1"
        src="image.jpg"
      />
      <p
        class="text-xs"
      >
        Product Name
      </p>
    </article>
  </div>
</div>
`;
